appserver/transaction/jta/src/main/java/com/sun/enterprise/transaction/JavaEETransactionImpl.java

```diff
@@ -679,29 +721,32 @@ public void registerSynchronization(Synchronization sync)
         // END OF IASRI 4660742
 
         checkTransationActive();
-        if ( jtsTx != null )
-            jtsTx.registerSynchronization(sync);
-        else
+        if (jtsTx == null) {
             syncs.add(sync);
+        } else {
+            jtsTx.registerSynchronization(sync);
+        }
     }
 
+    @Override
     public void setRollbackOnly() throws IllegalStateException, SystemException {
         checkTransationActive();
-        if ( jtsTx != null )
```

Comment:
Say no to if/else in negation!

Reply:
It all depends. Readability is not about following a set of rules. Its a property of the whole. There might be cases where a negation reads better and nobody should start asking to apply some rule. Here on the other hand flipping makes it more readable :)

Reply:
This is not the case. Maybe I should use smileys :)
Usually negation is maybe most usual cause of misunderstandings. Especially when I am tired and it is used in a sequence of ternary operators ...